src/os/bluestore/NVMEDevice.cc

```diff
@@ -266,12 +266,33 @@ class SharedDriverData {
 	return queues.at(i%queue_number);
   }
 
-  void register_device(NVMEDevice *device) {
+  int register_device(NVMEDevice *device) {
     // in case of registered_devices, we stop thread now.
     // Because release is really a rare case, we could bear this
     _aio_stop();
+
+     for (auto &&it : registered_devices) {
+      uint64_t offset = device->offset;
+      uint64_t size = device->get_size();
+      uint64_t offset_it = it->offset;
+      uint64_t size_it = it->get_size();
+
+      if (offset == offset_it) {
```

Comment:
do we allow offset == offset_it in any case? In my opinion, we don't allow any two devices totally overlap.

Reply:
Will same file be opened twice?

Reply:
yeah……that will be happened For block device, agree